File: 938. Range Sum of BST.java
  //Definition for a binary tree node.
  public class TreeNode {
      int val;
      TreeNode left;
      TreeNode right;
      TreeNode() {}
      TreeNode(int val) { this.val = val; }
      TreeNode(int val, TreeNode left, TreeNode right) {
          this.val = val;
          this.left = left;
          this.right = right;
 }
 }
class Solution {
    int answer;
    public int rangeSumBST(TreeNode root, int L, int R) {
        answer = 0;
        TreeNode c = root;
        addition(c, L, R);
        return answer;
    }
    
    public void addition(TreeNode n, int L, int R){ //recursively go through                                                the BST and add when appropriate
        if(n != null){
            if(n.val >= L && n.val <= R)
                answer += n.val;
            if(n.val < R)
                addition(n.right, L, R);
            if(L < n.val)
                addition(n.left, L, R);
        }
    }
}

File: 938. Range Sum of BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        self.range_sum = 0
        
        def traverse(node):
            if node:
                if node.val >= low and node.val <= high:
                    self.range_sum += node.val
                
                if node.left:
                    traverse(node.left)
                if node.right:
                    traverse(node.right)
            else:
                return
        
        traverse(root)
        return self.range_sum